tests/integ_tests/workflow_tests/local_activities.rs

```diff
@@ -242,12 +244,31 @@ async fn cancel_immediate(#[case] cancel_type: ActivityCancellationType) {
 
 struct LACancellerInterceptor {
     token: CancellationToken,
+    cancel_on_workflow_completed: bool,
 }
 #[async_trait::async_trait(?Send)]
 impl WorkerInterceptor for LACancellerInterceptor {
-    async fn on_workflow_activation_completion(&self, _: &WorkflowActivationCompletion) {}
+    async fn on_workflow_activation_completion(&self, completion: &WorkflowActivationCompletion) {
+        if !self.cancel_on_workflow_completed {
+            return;
+        }
+        if let Some(workflow_activation_completion::Status::Successful(
+            workflow_completion::Success { commands, .. },
+        )) = completion.status.as_ref()
+        {
+            if !commands.is_empty() {
+                if let Some(Variant::CompleteWorkflowExecution(_)) =
+                    commands[commands.len() - 1].variant.as_ref()
```

Comment:
`.last().unwrap()` is a bit more idiomatic

But I think this whole thing may not be needed. Just call the manual cancel token right at the end of the workflow function.

Reply:
Changed to use `last` and got rid of the `is_empty` check.